packages/atomic/src/components/commerce/product-template-components/atomic-product-description/atomic-product-description.tsx

```diff
@@ -0,0 +1,134 @@
+import {Product} from '@coveo/headless/commerce';
+import {Component, State, h, Element, Prop} from '@stencil/core';
+import ArrowDown from '../../../../images/arrow-down.svg';
+import {getFieldValueCaption} from '../../../../utils/field-utils';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Button} from '../../../common/button';
+import {CommerceBindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import {ProductContext} from '../product-template-decorators';
+
+/**
+ * @internal
+ * The `atomic-product-description` component renders the description of a product.
+ */
+@Component({
+  tag: 'atomic-product-description',
+  styleUrl: 'atomic-product-description.pcss',
+  shadow: false,
+})
+export class AtomicProductDescription
+  implements InitializableComponent<CommerceBindings>
+{
+  @InitializeBindings() public bindings!: CommerceBindings;
+  @ProductContext() private product!: Product;
+
+  @Element() hostElement!: HTMLElement;
+
+  public error!: Error;
+
+  @State() private isExpanded = false;
+  @State() private isTruncated = false;
+
+  private descriptionText!: HTMLDivElement;
+  private resizeObserver!: ResizeObserver;
+
+  /**
+   * The number of lines after which the product description should be truncated.
+   */
+  @Prop() public truncateAfter?: 1 | 2 | 3 | 4 = 2;
+
+  /**
+   * Whether the product description should be truncated.
+   */
+  @Prop() public shouldTruncate: boolean = true;
+
+  constructor() {
+    this.resizeObserver = new ResizeObserver(() => {
+      if (
+        this.descriptionText &&
+        this.descriptionText.scrollHeight > this.descriptionText.clientHeight
+      ) {
+        this.isTruncated = true;
+      } else {
+        this.isTruncated = false;
+      }
+    });
+  }
+
+  componentDidLoad() {
+    this.descriptionText = this.hostElement.querySelector(
+      '.product-description-text'
+    ) as HTMLDivElement;
+    if (this.descriptionText) {
+      this.resizeObserver.observe(this.descriptionText);
+    }
+  }
+
+  private onToggleExpand() {
+    this.isExpanded = !this.isExpanded;
+  }
+
+  private getLineClampClass() {
+    const lineClampMap = {
+      1: 'line-clamp-1',
+      3: 'line-clamp-3',
+      4: 'line-clamp-4',
+    };
+    return (
+      lineClampMap[this.truncateAfter as keyof typeof lineClampMap] ||
+      'line-clamp-2'
+    );
+  }
```

Comment:
Needed to ensure the tailwind classes don't get purged. Alternatively, could safelist them...